components/user-mgt/org.wso2.carbon.user.mgt/src/test/java/org/wso2/carbon/user/mgt/UserRealmProxyTest.java

```diff
@@ -0,0 +1,277 @@
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.user.mgt;
+
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.CarbonConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.registry.api.Registry;
+import org.wso2.carbon.user.api.Claim;
+import org.wso2.carbon.user.api.ClaimMapping;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.core.AuthorizationManager;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.claim.ClaimManager;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+import org.wso2.carbon.user.core.hybrid.HybridRoleManager;
+import org.wso2.carbon.user.mgt.common.ClaimValue;
+import org.wso2.carbon.user.mgt.common.FlaggedName;
+import org.wso2.carbon.user.mgt.common.UserAdminException;
+import org.wso2.carbon.user.mgt.common.UserRealmInfo;
+
+import java.lang.reflect.Field;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.testng.Assert.assertEquals;
+import static org.wso2.carbon.user.mgt.UserRealmProxy.PERMISSION;
+
+@PowerMockIgnore({"javax.net.*", "javax.security.*", "javax.crypto.*", "javax.xml.*"})
+@PrepareForTest({UserRealm.class, AuthorizationManager.class, AbstractUserStoreManager.class})
+public class UserRealmProxyTest {
+    private UserRealm realm;
+    private UserRealmProxy userRealmProxy;
+    private UserStoreManager userStoreManagerWithAb;
+    private UserStoreManager userStoreManager;
+    private AuthorizationManager authorizationManager;
+    private ClaimManager claimManager;
+
+    @DataProvider(name = "userListCount")
+    public static Object[][] itrCount() {
+        return new Object[][]{{10, 3}, {0, 1}};
+    }
+
+    @BeforeTest
+    public void setUp() throws Exception {
+        startTenantFlow("carbon.super");
+        realm = mock(UserRealm.class);
+        userStoreManagerWithAb = mock(AbstractUserStoreManager.class);
+        userStoreManager = mock(UserStoreManager.class);
+        authorizationManager = mock(AuthorizationManager.class);
+        claimManager = mock(ClaimManager.class);
+        userRealmProxy = new UserRealmProxy(realm);
+        Mockito.when(realm.getRealmConfiguration()).thenReturn(this.getSampleRelaimConfiguration());
```

Comment:
Better to use static import Mockito.* as the others.

Reply:
Need to use PoertMockito as well in other methods which are about to commit. So i need both mockito.when and powerMock.when